File: src/CorporateMessengerWithAgile/Application/Dto/Summary/EmployeeSummaryDto.cs
namespace Application.Dto.Summary
{
    public class EmployeeSummaryDto : BaseDto
    {
        public Guid CompanyId { get; set; }
        public Guid PositionInCompanyId { get; set; }
        public Guid UserId { get; set; }
        public IReadOnlyList<Guid> TeamMemberIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/EmployeeSummaryDto.ts
// src/models/entity/EmployeeSummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const EmployeeSummaryDtoSchema = BaseDtoSchema.extend({
    companyId: GuidSchema,
    positionInCompanyId: GuidSchema,
    userId: GuidSchema,
    teamMemberIds: z.array(GuidSchema).optional(),
});

export type EmployeeSummaryDto = z.infer<typeof EmployeeSummaryDtoSchema>;
